### PR TITLE
[BUGFIX] Show page tree according to DB Mounts (#411)

### PR DESCRIPTION
**Pull request by cdaecke**
With this fix the page tree will be shown correctly according to the DB Mounts of an user.

See issue #411

### CODE REVIEW

File: Classes/Hook/FrontendEditingInitializationHook.php [hunk 1]
```diff
@@ -466,15 +471,96 @@ protected function getAbsolutePath(string $path): string
      */
     protected function getPageTreeStructure(): array
     {
+        $entryPoints = $this->getAllEntryPointPageTrees();
+
+        foreach ($entryPoints as $entryPoint) {
+            if ($entryPoint['uid'] == 0) {
```
Comment (MattiasNilsson) [inline Classes/Hook/FrontendEditingInitializationHook.php:477]: Do an absolute comparison. 

`if ($entryPoint['uid'] === 0) {`

File: Classes/Hook/FrontendEditingInitializationHook.php [hunk 2]
```diff
@@ -466,15 +471,96 @@ protected function getAbsolutePath(string $path): string
...
+        $repository = GeneralUtility::makeInstance(PageTreeRepository::class, (int)$backendUser->workspace, [], $additionalPageTreeQueryRestrictions);
+
+        $entryPoints = (int)($backendUser->uc['pageTree_temporaryMountPoint'] ?? 0);
+        if ($entryPoints > 0) {
+            $entryPoints = [$entryPoints];
+        } else {
+            $entryPoints = array_map('intval', $backendUser->returnWebmounts());
+            $entryPoints = array_unique($entryPoints);
+            if (empty($entryPoints)) {
+                // use a virtual root
```
Comment (MattiasNilsson) [inline Classes/Hook/FrontendEditingInitializationHook.php:520]: Start comments with a Uppercase as when building a sentence.

File: Classes/Hook/FrontendEditingInitializationHook.php [hunk 3]
```diff
@@ -466,15 +471,96 @@ protected function getAbsolutePath(string $path): string
...
+                // since those will be the "sub pages" of the virtual root
+                $entryPoints = [0];
+            }
+        }
+        if (empty($entryPoints)) {
+            return [];
+        }
+
+        foreach ($entryPoints as $k => &$entryPoint) {
+            /*
```
Comment (MattiasNilsson) [inline Classes/Hook/FrontendEditingInitializationHook.php:531]: Remove the commented code.

File: Classes/Hook/FrontendEditingInitializationHook.php [hunk 4]
```diff
@@ -466,15 +471,96 @@ protected function getAbsolutePath(string $path): string
...
+                foreach ($entryPointRootLine as $rootLineEntry) {
+                    $parentUid = $rootLineEntry['uid'];
+                    if (!empty($this->backgroundColors[$parentUid]) && empty($this->backgroundColors[$entryPoint])) {
+                        $this->backgroundColors[$entryPoint] = $this->backgroundColors[$parentUid];
+                    }
+                }
+            }
+
+            $entryPoint = $repository->getTree($entryPoint, function ($page) use ($backendUser) {
+                // check each page if the user has permission to access it
```
Comment (MattiasNilsson) [inline Classes/Hook/FrontendEditingInitializationHook.php:553]: Start comments with a Uppercase as when building a sentence.

File: Classes/Hook/FrontendEditingInitializationHook.php (hunk 3 above)
Comment (cdaecke) [reply Classes/Hook/FrontendEditingInitializationHook.php:531]: The reason, why I commented this out was, that I don't have the data for `$this->hiddenRecords`, but I was not sure, if this is somehow needed... What to do? Just remove it?

Comment (cdaecke) [reply Classes/Hook/FrontendEditingInitializationHook.php:531]: One more information on this: As stated in the comments to the method `getAllEntryPointPageTrees()`, the code was taken form the core.

Comment (cdaecke) [reply Classes/Hook/FrontendEditingInitializationHook.php]: @MattiasNilsson Checking the code today again, I was able to fix this.

**Pull request merged**